### ISSUE TITLE
Rule `new-cap` does not report when capitalized function is not constructor

### ISSUE THREAD

**Issue by roadhump** [eslint/eslint#904]
JSHint `newcap` rule reports when capitalized function is called directly

``` js
var Foo = function(){};
Foo(); // Missing 'new' prefix when invoking a constructor.
```

It seems ESLint `new-cap` rule doesn't check this issue. 

It would be useful and consistent with JSHint. I suggest to make it as option, something like `new-cap: [2, 'strict']` to check such issues.
